### DISCUSSION TITLE
use SelenideElement in CollectionCondition

### DISCUSSION THREAD

**Discussion by vlad-velichko** [selenide/selenide#1757] [Ideas]
Thanks a lot for a wonderful tool.

Could you please change CollectionCondition allMatch, noneMatch, anyMatch to use SelenideElement instead of WebElement in predicate?

Warm regards,
Vlad

**Comment by asolntsev** [comment]
@vlad-velichko but why?

**Reply by vlad-velichko** [reply]
to use selenide methods like $
collection.shouldHave(allMatch(it->it.**$**["./my-selector"]).has(text(expectedText)))

**Reply by asolntsev** [reply]
@vlad-velichko But isn't it the same 
as `collection.shouldHave(allMatch(it->it.$["./my-selector"]).text().equals(expectedText))`? 

The problem with your suggestion is that `SelenideElement` has methods which _can wait_. And `$$.should*()` methods also _can wait_. It's quite unclear who should wait if you call `$$.shouldHave(allMatch(it -> it.shouldHave(text(...)))` - either external `$$.shouldHave` or internal `$.shouldHave`? Or both?

**Comment by asolntsev** [comment]
To summarize:
We rather are not going to make such a change.

It seems a dangerous idea to give users possibility of using "waiting" methods inside of a condition check which in turn is "waiting" method.

**Reply by vlad-velichko** [reply]
Got it. Thanks so much for the reply with a good explanation.
